polyhook2/Virtuals/VTableSwapHook.hpp

```diff
@@ -40,5 +41,64 @@ class VTableSwapHook : public PLH::IHook {
 	VFuncMap m_origVFuncs;
 	bool m_Hooked;
 };
+
+
+// storage class for address of a virtual function
+// also stores the function pointer type and index number on the class level
+template<uint16_t I, typename FuncPtr>
+struct VFunc {
+	VFunc() : func(nullptr) {};
+	VFunc(FuncPtr f) : func(f) {};
+	const FuncPtr func;
+	static const uint16_t func_index;
+	typedef FuncPtr func_type;
+};
+
+// definition of constant must reside outside class declaration
+template<uint16_t I, typename FuncPtr> const uint16_t VFunc<I, FuncPtr>::func_index = I;
+
+namespace detail {
+
+	// helper function to convert sequence of VFunc structs into a VFuncMap
+	// using recursive template definition
+	inline PLH::VFuncMap make_vfunc_map()
+	{
+		return PLH::VFuncMap{ };
+	};
+
+	template<uint16_t I, typename FuncPtr, typename ... VFuncTypes>
+	inline PLH::VFuncMap make_vfunc_map(VFunc<I, FuncPtr> vfunc, VFuncTypes ... vfuncs)
```

Comment:
I would prefer if this function was instead implemented as a constructor on the VTableSwapHook class itself.

Reply:
Yes, I agree. I've now implemented this.